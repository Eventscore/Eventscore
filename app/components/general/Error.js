//<Error text={'Server is down'} />
import React, { PropTypes } from 'react';
import { View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

// Consts and Libs
import { AppStyles } from '@theme/';

// Components
import { Spacer, Text, Button } from '@ui/';

const Error = ({ text, tryAgain }) => (
  <View style={[AppStyles.container, AppStyles.containerCentered]}>
    <Icon name={'ios-alert-outline'} size={50} color={'#CCC'} />

    <Spacer size={10} />

    <Text style={AppStyles.textCenterAligned}>{text}</Text>

    <Spacer size={20} />

    {!!tryAgain &&
      <Button
        small
        outlined
        title={'Try again'}
        onPress={tryAgain}
      />
    }
  </View>
);

Error.propTypes = { text: PropTypes.string, tryAgain: PropTypes.func };
Error.defaultProps = { text: 'Woops, Something went wrong.', tryAgain: null };
Error.componentName = 'Error';

export default Error;
