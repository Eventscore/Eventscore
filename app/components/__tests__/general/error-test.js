/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import { Alert } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Error from '@components/general/Error';

it('Error renders correctly', () => {
  const tree = renderer.create(
    <Error text={'We can\'t find that'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Error w/ Try Again Button renders correctly', () => {
  const tree = renderer.create(
    <Error tryAgain={() => { Alert.alert('Hey there'); }} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
