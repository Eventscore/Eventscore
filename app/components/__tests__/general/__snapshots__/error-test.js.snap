// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#CCC",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 9,
        "right": 0,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    onPress={null}
    p={false}
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "HelveticaNeue",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 18,
        },
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    We can't find that
  </Text>
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 19,
        "right": 0,
      }
    }
  />
</View>
`;

exports[`Error w/ Try Again Button renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#CCC",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 9,
        "right": 0,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    onPress={null}
    p={false}
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "HelveticaNeue",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 18,
        },
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Woops, Something went wrong.
  </Text>
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 19,
        "right": 0,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": 15,
            "marginRight": 15,
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderRadius": 2,
          },
          false,
          Object {
            "padding": 12,
          },
          Object {
            "borderColor": "#0E4EF8",
            "borderWidth": 1,
            "marginLeft": 0,
            "marginRight": 0,
            "padding": 8,
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": undefined,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "#0E4EF8",
              },
              Object {
                "fontSize": 17.5,
              },
              Object {
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontWeight": "bold",
              },
              Object {
                "fontFamily": "HelveticaNeue",
              },
            ],
          ]
        }
      >
        Try again
      </Text>
    </View>
  </View>
</View>
`;
